### PR TITLE
Add configmap and secret count metrics

### PR DESCRIPTION
**Pull request by wenzstev**
### What does this PR do?

This pull request adds two additional metrics to the Kubernetes State check:

- `kubernetes_state.secret.count`
- `kubernetes_state.configmap.count`

These metrics count the number of secrets and configmaps respectively, and can be aggregated by the `namespace` tag. These metrics can be toggled with two new settings for the check:

- `enable_configmap_count`
- `enable_secret_count`

These default to false; the metrics won't be gathered unless they're set to true. The metrics are gathered in the same way as other aggregation metrics.

### Motivation

Jira card [here](https://datadoghq.atlassian.net/browse/CONT-3648?atlOrigin=eyJpIjoiN2JjMjBiYTMxMTgwNGJlMzlmYTUwYmRiYzUyNjc0ZjUiLCJwIjoiaiJ9). We have counts for other Kubernetes resources, but not these. 

### Possible Drawbacks / Trade-offs

The code currently checks for the presence of the `enable` values after potential aggregate metrics are found. There may be a better way to do this. 

### Describe how to test/QA your changes

1. Spin up a version of the agent with this check using Helm.
2. Manually edit the Cluster Agent's `ClusterRole` that is generated in order to ensure that the agent has permission to list `ConfigMap` and `Secret` resources. 
3. Confirm that the metric count matches the actual count of the given resource. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/cluster/ksm/kubernetes_state.go [hunk 1]
```diff
@@ -138,6 +138,12 @@ type KSMConfig struct {
 	// Can be useful when running the check as cluster check
 	LeaderSkip bool `yaml:"skip_leader_election"`
 
+	//EnableConfigMapCount allows for enabling or disabling the kubernetes_state.configmap.count metric
+	EnableConfigMapCount bool `yaml:"enable_configmap_count"`
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply pkg/collector/corechecks/cluster/ksm/kubernetes_state.go:142]: In addition I don't think we need these 2 variables. we should be able to trigger the creation of the metrics only if the `configmap` or `secret` is added of the list of monitored resources.

File: pkg/collector/corechecks/cluster/ksm/kubernetes_state.go [hunk 2]
```diff
@@ -503,6 +509,14 @@ func (k *KSMCheck) processMetrics(sender aggregator.Sender, metrics map[string][
...
 			// First check for aggregator, because the check use _labels metrics to aggregate values.
 			if aggregator, found := k.metricAggregators[metricFamily.Name]; found {
+				if metricFamily.Name == "kube_configmap_info" && !k.instance.EnableConfigMapCount {
+					log.Tracef("ConfigMap metric found but counting ConfigMaps is disabled.")
+					continue
+				}
+				if metricFamily.Name == "kube_secret_info" && !k.instance.EnableSecretCount {
+					log.Tracef("Secret metric found but counting Secrets is disabled.")
+					continue
+				}
```
Comment (clamoriniere) [inline pkg/collector/corechecks/cluster/ksm/kubernetes_state.go:519]: imo, this is not needed if we remove the 2 options as I suggested in a previous comment.

**Review by clamoriniere — COMMENTED** [review]
Add few comments.
the changes made in `pkg/collector/corechecks/cluster/ksm/kubernetes_state_aggregators.go` should be enough.

Please also plan a pull request to add the new metrics in https://github.com/DataDog/integrations-core/blob/master/kubernetes_state_core/metadata.csv 🙇

**Comment by wenzstev** [comment]
@clamoriniere Done. I removed all of the additional code from the main check class. One just needs to add the given collectors to the Kubernetes State Core configuration in order to get the metrics. Also updated the Helm chart PR to reflect this: [https://github.com/DataDog/helm-charts/pull/1034](https://github.com/DataDog/helm-charts/pull/1034/files)

**Review by estherk15 — APPROVED** [review]
Looks good for docs

**Review by clamoriniere — APPROVED** [review]

**Pull request closed without merging**